### PR TITLE
Update upgrade scripts

### PR DESCRIPTION
**Pull request by daveroga**
Update upgrade scripts with latest changes and taking into account `.env` for deployment addresses.

### CODE REVIEW

File: scripts/upgrade/state/state-upgrade.ts
Comment (AndriianChestnykh) [inline scripts/upgrade/state/state-upgrade.ts:1]: Any reason why we don't write/overwrite the output files in all upgrade scripts?

File: scripts/upgrade/state/state-upgrade.ts [hunk 1]
```diff
@@ -64,26 +27,65 @@ async function getSigners(useImpersonation: boolean): Promise<any> {
...
+    config.poseidon3ContractAddress,
+  ];
   await stateDeployHelper.upgradeState(
     await stateContract.getAddress(),
     true,
     true,
-  )
+    deployStrategy,
+    poseidonContracts,
+    config.smtLibContractAddress,
```
[the remark this answers is not in the document]
Comment (daveroga) [reply scripts/upgrade/state/state-upgrade.ts:85]: You mean passing some redeploySmtLib param to redeploy smtLib contract like we are doing with `redeployVerifier` or `redeployCrossChainProofValidator` params?

File: scripts/upgrade/state/state-upgrade.ts
Comment (daveroga) [reply scripts/upgrade/state/state-upgrade.ts:1]: No. I'll do it.

Comment (daveroga) [reply scripts/upgrade/state/state-upgrade.ts:1]: Done

File: helpers/StateContractMigrationHelper.ts [hunk 2]
```diff
@@ -141,13 +141,27 @@ export class StateContractMigrationHelper extends ContractMigrationSteps {
   @log
   async upgradeContract(
     stateContract: Contract,
-    redeployVerifier = true,
-    redeployCrossChainProofValidator = true,
+    opts: {
+      redeployVerifier: boolean;
```
Comment (AndriianChestnykh) [inline helpers/StateContractMigrationHelper.ts:145]: `redeployGroth16Verifier` is better to differentiate

File: helpers/constants.ts
Comment (AndriianChestnykh) [inline helpers/constants.ts:1]: I think it worth to change all constants to SCREAMING_SNAKE_CASE according popular conventions. But only exported ones.

File: helpers/StateContractMigrationHelper.ts (hunk 2 above)
Comment (daveroga) [reply helpers/StateContractMigrationHelper.ts:145]: Yes, sure.

File: helpers/constants.ts
Comment (daveroga) [reply helpers/constants.ts:1]: Done

**Review by AndriianChestnykh — APPROVED** [review]

**Pull request merged**